lib/validate-schemas.spec.js

```diff
@@ -24,19 +24,157 @@ function loadRemoteSchema(uri) {
     });
 }
 
+const dataKeyTopicRegex = /^[a-zA-Z]([a-zA-Z0-9]+|\.|_)*[a-zA-Z0-9]$/;
+const dataKeyTopicMinLength = 0;
+const dataKeyTopicMaxLength = 64;
+const validatePropertyName = (errors, dataKey) => {
+  if (dataKey.length < dataKeyTopicMinLength) {
+    errors.push(`dataKey '${dataKey}' too short`);
+  }
+
+  if (dataKey.length > dataKeyTopicMaxLength) {
+    errors.push(`dataKey '${dataKey}' too long`);
+  }
+
+  if (!dataKeyTopicRegex.test(dataKey)) {
+    errors.push(`dataKey '${dataKey}' does not match convention`);
+  }
+};
+
+const validateTopic = (filepath, errors, topic) => {
+  if (topic === undefined) {
+    errors.push(`topic not set for ${filepath}`);
+    return;
+  }
+
+  if (topic.length < dataKeyTopicMinLength) {
+    errors.push(`topic '${topic}' too short`);
+  }
+
+  if (topic.length > dataKeyTopicMaxLength) {
+    errors.push(`topic '${topic}' too long`);
+  }
+
+  if (!dataKeyTopicRegex.test(topic)) {
+    errors.push(`topic '${topic}' does not match convention`);
+  }
+};
+
+const validateProcessingType = (filepath, errors, processingType) => {
+  if (processingType === undefined) {
+    errors.push(`processingType not set for ${filepath}`);
+    return;
+  }
+
+  if (
+    processingType !== "uplink_decoder" &&
+    processingType !== "downlink_encoder"
+  ) {
+    errors.push(
+      `processingType not one of ['uplink_decoder', 'downlink_encoder' for ${filepath}`,
+    );
+  }
+};
+
+const validateSchemaProperties = (errors, schema) => {
+  if (schema === undefined) {
+    return;
+  }
+
+  const schemaKeys = Object.keys(schema);
+  const allowedProperties = [
```

Comment:
There are more properties defined by the json schema standard that we might want to allow:
https://json-schema.org/draft/2020-12/json-schema-validation#name-a-vocabulary-for-structural

Reply:
can be added on demand 👍